pkg/bisect: TestBisectionResults failed

**Describe the bug**
https://github.com/google/syzkaller/actions/runs/24847952654/job/72740198079

```

2026-05-04T09:42:10.6623930Z --- FAIL: TestBisectionResults (0.15s)
2026-05-04T09:42:10.6624208Z     --- FAIL: TestBisectionResults/group (0.15s)
2026-05-04T09:42:10.6624574Z         git_test_util.go:122: "400"'s hash is f0b44ec9f7e5c92e66825bc0ae48898a10ecb4a0
2026-05-04T09:42:10.6645120Z ##[error]"400"'s hash is f0b44ec9f7e5c92e66825bc0ae48898a10ecb4a0
2026-05-04T09:42:10.6652894Z         git_test_util.go:122: "400"'s hash is af460c54eaed1ac1bdd596f4d73ae64827a4c698
2026-05-04T09:42:10.6654019Z ##[error]"400"'s hash is af460c54eaed1ac1bdd596f4d73ae64827a4c698
2026-05-04T09:42:10.6654961Z         git_test_util.go:117: failed to run ["git" "commit" "--allow-empty" "-m" "400"]: exit status 128
2026-05-04T09:42:10.6655975Z ##[error]failed to run ["git" "commit" "--allow-empty" "-m" "400"]: exit status 128
2026-05-04T09:42:10.6656764Z         --- FAIL: TestBisectionResults/group/fix-finds-fix (0.14s)
2026-05-04T09:42:10.6657177Z         git_test_util.go:122: "400"'s hash is b1c59603fa7267380f27e2e0f1ba8438917fe602
2026-05-04T09:42:10.6658450Z ##[error]"400"'s hash is b1c59603fa7267380f27e2e0f1ba8438917fe602
2026-05-04T09:42:10.6659405Z         git_test_util.go:122: "400"'s hash is 10bdc1cf638efb52e12a424a530deab5c89445f7
2026-05-04T09:42:10.6660224Z ##[error]"400"'s hash is 10bdc1cf638efb52e12a424a530deab5c89445f7
2026-05-04T09:42:10.6660987Z         git_test_util.go:122: "400"'s hash is 5cd96c4cc2bb6f27cf379d0841e0fa88766e944b
2026-05-04T09:42:10.6661780Z ##[error]"400"'s hash is 5cd96c4cc2bb6f27cf379d0841e0fa88766e944b
2026-05-04T09:42:10.6662557Z         git_test_util.go:122: "400"'s hash is c460bfc2c23b68c44d535ed72a8a6113cc5d22b4
2026-05-04T09:42:10.6663355Z ##[error]"400"'s hash is c460bfc2c23b68c44d535ed72a8a6113cc5d22b4
2026-05-04T09:43:08.2735498Z         git_test_util.go:122: "400"'s hash is 1da030bfad9b9e735c816997d9a0df5d3e64fa74
2026-05-04T09:43:08.2736828Z ##[error]"400"'s hash is 1da030bfad9b9e735c816997d9a0df5d3e64fa74
2026-05-04T09:43:08.2742529Z         git_test_util.go:120: failed to run ["git" "log" "--format=%H%n%s%n%ae%n%an%n%ad%n%P%n%cd%n%b---===syzkaller-patch-separator===---" "-n" "1" "-p" "-U0" "HEAD"]: exit status 128
2026-05-04T09:43:08.2744303Z ##[error]failed to run ["git" "log" "--format=%H%n%s%n%ae%n%an%n%ad%n%P%n%cd%n%b---===syzkaller-patch-separator===---" "-n" "1" "-p" "-U0" "HEAD"]: exit status 128
2026-05-04T09:43:08.2745491Z         --- FAIL: TestBisectionResults/group/fix-does-not-repro (0.15s)
2026-05-04T09:43:08.2745880Z panic: test executed panic(nil) or runtime.Goexit
2026-05-04T09:43:08.2746091Z 
2026-05-04T09:43:08.2746175Z goroutine 195 [running]:
2026-05-04T09:43:08.2746405Z testing.tRunner.func1.2({0x26d8d40, 0x4922a80})
2026-05-04T09:43:08.2746702Z 	/usr/local/go/src/testing/testing.go:1974 +0x419
2026-05-04T09:43:08.2746968Z testing.tRunner.func1()
2026-05-04T09:43:08.2747192Z 	/usr/local/go/src/testing/testing.go:1977 +0x67f
2026-05-04T09:43:08.2747438Z runtime.Goexit()
2026-05-04T09:43:08.2747904Z 	/usr/local/go/src/runtime/panic.go:694 +0x5e
2026-05-04T09:43:08.2748197Z testing.(*common).FailNow(0xc00094dd48)
2026-05-04T09:43:08.2748515Z 	/usr/local/go/src/testing/testing.go:1022 +0x7b
2026-05-04T09:43:08.2748841Z testing.(*common).Fatal(0xc00094dd48, {0xc000f0fa90, 0x1, 0x1})
2026-05-04T09:43:08.2749164Z 	/usr/local/go/src/testing/testing.go:1221 +0x85
2026-05-04T09:43:08.2750221Z github.com/google/syzkaller/pkg/vcs.(*TestRepo).CommitChangeset(0xc000d8c300, {0xc0002e4cdd, 0x3}, {0x0, 0x0, 0x0?})
2026-05-04T09:43:08.2750928Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/pkg/vcs/git_test_util.go:120 +0x451
2026-05-04T09:43:08.2751461Z github.com/google/syzkaller/pkg/vcs.(*TestRepo).CommitChange(...)
2026-05-04T09:43:08.2751977Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/pkg/vcs/git_test_util.go:81
2026-05-04T09:43:08.2752483Z github.com/google/syzkaller/pkg/bisect.createTestRepo(0xc00094dd48)
2026-05-04T09:43:08.2753017Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/pkg/bisect/bisect_test.go:185 +0x61c
2026-05-04T09:43:08.2753602Z github.com/google/syzkaller/pkg/bisect.TestBisectionResults.func1.1(0xc000f89b08)
2026-05-04T09:43:08.2754185Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/pkg/bisect/bisect_test.go:744 +0x188
2026-05-04T09:43:08.2754640Z testing.tRunner(0xc000f89b08, 0xc000142f60)
2026-05-04T09:43:08.2754922Z 	/usr/local/go/src/testing/testing.go:2036 +0x21d
2026-05-04T09:43:08.2755200Z created by testing.(*T).Run in goroutine 79
2026-05-04T09:43:08.2755461Z 	/usr/local/go/src/testing/testing.go:2101 +0xb13
2026-05-04T09:43:08.2755707Z exit status 2
2026-05-04T09:43:08.2755910Z FAIL	github.com/google/syzkaller/pkg/bisect	2.946s
```

```
--- FAIL: TestBisectionResults (12.04s)
    --- FAIL: TestBisectionResults/group (0.27s)
        --- FAIL: TestBisectionResults/group/fix-cross-tree (0.93s)
            git_test_util.go:122: "400"'s hash is 82f0fa96fca8d2621f0ef84002ece6282ab6782b
Error: "400"'s hash is 82f0fa96fca8d2621f0ef84002ece6282ab6782b
            git_test_util.go:122: "401"'s hash is 65b4ed94529959ef0ad316259cb252eeb44d53eb
Error: "401"'s hash is 65b4ed94529959ef0ad316259cb252eeb44d53eb
            git_test_util.go:122: "402"'s hash is ef58e8d015fe450e38f77a45552411deb9cf38be
Error: "402"'s hash is ef58e8d015fe450e38f77a45552411deb9cf38be
            git_test_util.go:122: "403"'s hash is 343cfe558e61e6e252c1aaccb8a46edd74bcfbfd
Error: "403"'s hash is 343cfe558e61e6e252c1aaccb8a46edd74bcfbfd
            git_test_util.go:122: "404"'s hash is b50d343dd898189abc8a38729e84b7235e49229f
Error: "404"'s hash is b50d343dd898189abc8a38729e84b7235e49229f
            git_test_util.go:122: "405"'s hash is 1bfa6579c86c1ef7b829126c0dad6528c9fa5cf6
Error: "405"'s hash is 1bfa6579c86c1ef7b829126c0dad6528c9fa5cf6
            git_test_util.go:122: "500"'s hash is 1769ea802c34e65e03064a4a572bf9ae2afbf049
Error: "500"'s hash is 1769ea802c34e65e03064a4a572bf9ae2afbf049
            git_test_util.go:122: "501"'s hash is ef95324e21c843441baf0102dbba856320549c84
Error: "501"'s hash is ef95324e21c843441baf0102dbba856320549c84
            git_test_util.go:122: "502"'s hash is 52c915c63e78ee3619a81a276e3cd0996cb8fd53
Error: "502"'s hash is 52c915c63e78ee3619a81a276e3cd0996cb8fd53
            git_test_util.go:122: "503"'s hash is 2cef535a8a8497adbee7ef1e02c44ab44d1f1708
Error: "503"'s hash is 2cef535a8a8497adbee7ef1e02c44ab44d1f1708
            git_test_util.go:122: "504"'s hash is 17294548cfa7f17ff38df54c5790a79096ddc597
Error: "504"'s hash is 17294548cfa7f17ff38df54c5790a79096ddc597
            git_test_util.go:122: "505"'s hash is c6516b24d9160c61670c90b35fc3d330a6508dfb
Error: "505"'s hash is c6516b24d9160c61670c90b35fc3d330a6508dfb
            git_test_util.go:122: "600"'s hash is 8ddc4a8383bfc155f31362539aaee795b5ac6d95
Error: "600"'s hash is 8ddc4a8383bfc155f31362539aaee795b5ac6d95
            git_test_util.go:122: "601"'s hash is 5ff5d96977fe9de0d21adb4a09b931db55092d0b
Error: "601"'s hash is 5ff5d96977fe9de0d21adb4a09b931db55092d0b
            git_test_util.go:122: "602"'s hash is 7364c00bb7be05967ba98d5a44c1d8a2d1d02f6b
Error: "602"'s hash is 7364c00bb7be05967ba98d5a44c1d8a2d1d02f6b
            git_test_util.go:120: failed to run ["git" "-c" "core.hooksPath=/dev/null" "log" "--format=%H%n%s%n%ae%n%an%n%ad%n%P%n%cd%n%b---===syzkaller-patch-separator===---" "-n" "1" "-p" "-U0" "HEAD"]: exit status 128
Error: failed to run ["git" "-c" "core.hooksPath=/dev/null" "log" "--format=%H%n%s%n%ae%n%an%n%ad%n%P%n%cd%n%b---===syzkaller-patch-separator===---" "-n" "1" "-p" "-U0" "HEAD"]: exit status 128
FAIL
coverage: 82.8% of statements
```

Now with more logging:
```
2026-05-06T14:01:23.1896998Z --- FAIL: TestBisectionResults (11.24s)
2026-05-06T14:01:23.1897494Z     --- FAIL: TestBisectionResults/group (0.24s)
2026-05-06T14:01:23.1898289Z         --- FAIL: TestBisectionResults/group/fix-HEAD-broken (1.89s)
2026-05-06T14:01:23.1899042Z             git_test_util.go:122: "400"'s hash is 5c0caac67bce8cf4983d8ce3af02bee8264dcdef
2026-05-06T14:01:23.1933368Z ##[error]"400"'s hash is 5c0caac67bce8cf4983d8ce3af02bee8264dcdef
2026-05-06T14:01:23.1943828Z             git_test_util.go:122: "401"'s hash is 83861cb6b8694f98f9817c147620009d108990d2
2026-05-06T14:01:23.1945915Z ##[error]"401"'s hash is 83861cb6b8694f98f9817c147620009d108990d2
2026-05-06T14:01:23.1958123Z             git_test_util.go:122: "402"'s hash is e9df24bfbff25c8b8e3b76c8cba59b7fd793dc73
2026-05-06T14:01:23.1959780Z ##[error]"402"'s hash is e9df24bfbff25c8b8e3b76c8cba59b7fd793dc73
2026-05-06T14:01:23.1961077Z             git_test_util.go:122: "403"'s hash is ce861069ebc739a52ac11c25f4beca952f35eccf
2026-05-06T14:01:23.1962250Z ##[error]"403"'s hash is ce861069ebc739a52ac11c25f4beca952f35eccf
2026-05-06T14:01:23.1963573Z             git_test_util.go:122: "404"'s hash is 1fad3f0c700a66e8126a70d89e1cf161b3c0bae4
2026-05-06T14:01:23.1964973Z ##[error]"404"'s hash is 1fad3f0c700a66e8126a70d89e1cf161b3c0bae4
2026-05-06T14:01:23.1966767Z             git_test_util.go:122: "405"'s hash is 3b9901a7448ed3fc66ce5eb68903f7ca2f46f461
2026-05-06T14:01:23.1968180Z ##[error]"405"'s hash is 3b9901a7448ed3fc66ce5eb68903f7ca2f46f461
2026-05-06T14:01:23.1969703Z             git_test_util.go:122: "500"'s hash is cdca00b936c7e775fd907b6cc5e70bdd51a16432
2026-05-06T14:01:23.1971184Z ##[error]"500"'s hash is cdca00b936c7e775fd907b6cc5e70bdd51a16432
2026-05-06T14:01:23.1972362Z             git_test_util.go:122: "501"'s hash is c45de98125cb1627703de7679031080c960dd9e9
2026-05-06T14:01:23.1973596Z ##[error]"501"'s hash is c45de98125cb1627703de7679031080c960dd9e9
2026-05-06T14:01:23.1975034Z             git_test_util.go:122: "502"'s hash is 5f943d00ea904fa6838d9e59a69533da4372ea88
2026-05-06T14:01:23.1976302Z ##[error]"502"'s hash is 5f943d00ea904fa6838d9e59a69533da4372ea88
2026-05-06T14:01:23.1977608Z             git_test_util.go:122: "503"'s hash is 975d84d2e10c45bc3be5bc3808c4da230d14f34f
2026-05-06T14:01:23.1979219Z ##[error]"503"'s hash is 975d84d2e10c45bc3be5bc3808c4da230d14f34f
2026-05-06T14:01:23.1980608Z             git_test_util.go:122: "504"'s hash is 0bc7dda267646de0e65831efe74cfdaba69c5350
2026-05-06T14:01:23.1981938Z ##[error]"504"'s hash is 0bc7dda267646de0e65831efe74cfdaba69c5350
2026-05-06T14:01:23.1983084Z             git_test_util.go:122: "505"'s hash is d122e2b0d2d8ad9daa2bf00610907c4f472c49d5
2026-05-06T14:01:23.1984595Z ##[error]"505"'s hash is d122e2b0d2d8ad9daa2bf00610907c4f472c49d5
2026-05-06T14:01:23.1985867Z             git_test_util.go:122: "600"'s hash is e6e35a85a5a5ffbbb66320732d7339c3f5a8420f
2026-05-06T14:01:23.1986995Z ##[error]"600"'s hash is e6e35a85a5a5ffbbb66320732d7339c3f5a8420f
2026-05-06T14:01:23.2029436Z             git_test_util.go:122: "601"'s hash is 9dae7286c7078907d2ba97ea9cb1130d79bcd4a5
2026-05-06T14:01:23.2031485Z ##[error]"601"'s hash is 9dae7286c7078907d2ba97ea9cb1130d79bcd4a5
2026-05-06T14:01:23.2033242Z             git_test_util.go:122: "602"'s hash is f7dfe82523d6010013a8e6eb34f67b706c86ff35
2026-05-06T14:01:23.2034704Z ##[error]"602"'s hash is f7dfe82523d6010013a8e6eb34f67b706c86ff35
2026-05-06T14:01:23.2036376Z             git_test_util.go:122: "603"'s hash is 7982c60e8088b18432b2537db017c2686b3fe7bd
2026-05-06T14:01:23.2037798Z ##[error]"603"'s hash is 7982c60e8088b18432b2537db017c2686b3fe7bd
2026-05-06T14:01:23.2039572Z             git_test_util.go:122: "604"'s hash is 8c5961bb93916483e609e8a72eeca60f3c31754d
2026-05-06T14:01:23.2041010Z ##[error]"604"'s hash is 8c5961bb93916483e609e8a72eeca60f3c31754d
2026-05-06T14:01:23.2042412Z             git_test_util.go:122: "605"'s hash is 13b87920eef6554ec937b536bb6f996bf294c63b
2026-05-06T14:01:23.2043795Z ##[error]"605"'s hash is 13b87920eef6554ec937b536bb6f996bf294c63b
2026-05-06T14:01:23.2045194Z             git_test_util.go:122: "650"'s hash is 6e7eee967a1cecb6cb70b44047670e05844996dc
2026-05-06T14:01:23.2046564Z ##[error]"650"'s hash is 6e7eee967a1cecb6cb70b44047670e05844996dc
2026-05-06T14:01:23.2048146Z             git_test_util.go:122: "701"'s hash is c9f41ba5e2134842be88cbb72fc74e5df45d5d9b
2026-05-06T14:01:23.2049534Z ##[error]"701"'s hash is c9f41ba5e2134842be88cbb72fc74e5df45d5d9b
2026-05-06T14:01:23.2050906Z             git_test_util.go:122: "702"'s hash is 91a79637cf4f37d7a1cd1940e2d7ce35c0b8ffd1
2026-05-06T14:01:23.2052333Z ##[error]"702"'s hash is 91a79637cf4f37d7a1cd1940e2d7ce35c0b8ffd1
2026-05-06T14:01:23.2053692Z             git_test_util.go:122: "703"'s hash is a4013e6a5c456aea5d1204e2d162069a45e72b25
2026-05-06T14:01:23.2055461Z ##[error]"703"'s hash is a4013e6a5c456aea5d1204e2d162069a45e72b25
2026-05-06T14:02:03.5620311Z             git_test_util.go:122: "704"'s hash is 6ea6b6edbdf3e82fa19d779684aea3209066cada
2026-05-06T14:02:03.5625973Z ##[error]"704"'s hash is 6ea6b6edbdf3e82fa19d779684aea3209066cada
2026-05-06T14:02:03.5628681Z             git_test_util.go:122: "705"'s hash is 0f5dabf1f0fb666894ba1131654c6250c604a91e
2026-05-06T14:02:03.5629699Z ##[error]"705"'s hash is 0f5dabf1f0fb666894ba1131654c6250c604a91e
2026-05-06T14:02:03.5630919Z             git_test_util.go:122: "800"'s hash is b1181177e848313c36c1fe733d9bbf7a7e98cfce
2026-05-06T14:02:03.5631850Z ##[error]"800"'s hash is b1181177e848313c36c1fe733d9bbf7a7e98cfce
2026-05-06T14:02:03.5632799Z             git_test_util.go:122: "801"'s hash is f42725e4ff14aeab2e4221a0a80eda9dcfea54ab
2026-05-06T14:02:03.5634207Z ##[error]"801"'s hash is f42725e4ff14aeab2e4221a0a80eda9dcfea54ab
2026-05-06T14:02:03.5635130Z             git_test_util.go:122: "802"'s hash is 1a761c89b7e2ab43d555e3aa091a0f5a6aaf55b8
2026-05-06T14:02:03.5636007Z ##[error]"802"'s hash is 1a761c89b7e2ab43d555e3aa091a0f5a6aaf55b8
2026-05-06T14:02:03.5636863Z             git_test_util.go:122: "803"'s hash is fec76bc489423d5c59e14328fcd390c7da4074d5
2026-05-06T14:02:03.5637712Z ##[error]"803"'s hash is fec76bc489423d5c59e14328fcd390c7da4074d5
2026-05-06T14:02:03.5638766Z             git_test_util.go:122: "790"'s hash is 21988a57bba3cf285e7e6e34a664744e0ca1ee63
2026-05-06T14:02:03.5639582Z ##[error]"790"'s hash is 21988a57bba3cf285e7e6e34a664744e0ca1ee63
2026-05-06T14:02:03.5640365Z             git_test_util.go:122: "791"'s hash is 070da803a75c370ce22c75d0809132b599b00572
2026-05-06T14:02:03.5641148Z ##[error]"791"'s hash is 070da803a75c370ce22c75d0809132b599b00572
2026-05-06T14:02:03.5641892Z             git_test_util.go:122: "792"'s hash is 170ef554479eb629e3ecc0203394b9e9a8df0e7c
2026-05-06T14:02:03.5642659Z ##[error]"792"'s hash is 170ef554479eb629e3ecc0203394b9e9a8df0e7c
2026-05-06T14:02:03.5643475Z             git_test_util.go:122: "805"'s hash is 8786352da3521622d17f6aaa4b0c09889b64674c
2026-05-06T14:02:03.5644232Z ##[error]"805"'s hash is 8786352da3521622d17f6aaa4b0c09889b64674c
2026-05-06T14:02:03.5644974Z             git_test_util.go:122: "900"'s hash is 3c331035b4b5a1fde49919261ada37699d00850f
2026-05-06T14:02:03.5645733Z ##[error]"900"'s hash is 3c331035b4b5a1fde49919261ada37699d00850f
2026-05-06T14:02:03.5646535Z             git_test_util.go:117: failed to run ["git" "commit" "--allow-empty" "-m" "901"]: exit status 128
2026-05-06T14:02:03.5647486Z ##[error]failed to run ["git" "commit" "--allow-empty" "-m" "901"]: exit status 128
2026-05-06T14:02:03.5648492Z                 error: unable to create temporary file: No such file or directory
2026-05-06T14:02:03.5648855Z                 fatal: failed to write commit object
2026-05-06T14:02:03.5649102Z                 
2026-05-06T14:02:03.5649256Z FAIL
2026-05-06T14:02:03.5649424Z coverage: 82.5% of statements
```

--- a/pkg/vcs/git_test_util.go
+++ b/pkg/vcs/git_test_util.go
@@ -56,6 +56,10 @@ func MakeTestRepo(t *testing.T, dir string) *TestRepo {
 	repo.Git("init")
 	repo.Git("config", "--add", "user.email", userEmail)
 	repo.Git("config", "--add", "user.name", userName)
+	repo.Git("config", "--add", "gc.auto", "0")
+	repo.Git("config", "--add", "gc.autoDetach", "false")
+	repo.Git("config", "--add", "maintenance.auto", "false")
+	repo.Git("config", "--add", "maintenance.autoDetach", "false")
 	return repo
 }
 

